src/shell/commands/exit.rs

```diff
@@ -21,9 +21,9 @@ impl ShellCommand for ExitCommand {
     let result = match execute_exit(context.args) {
       Ok(code) => ExecuteResult::Exit(code, Vec::new()),
       Err(err) => {
-        // even if parsing args fails `exit` always exits
+        // exit exits with exit code 2 on argument parse failure
```

Comment:
It seems the exact error code to use is not dictated, but generally exit codes for error parsing is 2.